src/main/java/frc/robot/subsystems/ball/BallImpl.java

```diff
@@ -0,0 +1,47 @@
+package frc.robot.subsystems.ball;
+
+import ca.team3161.lib.robot.subsystem.RepeatingPooledSubsystem;
+import java.util.concurrent.TimeUnit;
+
+import frc.robot.subsystems.ball.feeder.FeederImpl;
+import frc.robot.subsystems.ball.shooter.ShooterImpl;
+
+public class BallImpl extends RepeatingPooledSubsystem implements Ball {
+
+    ShooterImpl shooter;
+    FeederImpl feeder;
+    
+    BallImpl() {
+        // PLACEHOLDER
+        super(1, TimeUnit.SECONDS);
+        this.shooter = new ShooterImpl();
+        this.feeder = new FeederImpl();
+    }
+    
+    public void shoot(){
+        this.shooter.runShooter();
+        if(this.shooter.readyForBalls()){
+            this.feeder.feedBalls();
+        }
+    }
+
+    public void cancelShoot(){
+        this.shooter.stopShooter();
+        this.feeder.stop();
+    }
+
+    public void unload() {
+        this.feeder.unload();
+        // Seems kind of redundant, but abstraction
+    }
+
+    public void defineResources(){
+        return;
+        // IDK if I need to require the shooter and feeder or not
+    }
+
+    public void task() {
+        // PLACEHOLDER
+        return;
```

Comment:
This would be a place to implement some logic relating to `shoot`. Right now the `readyForBalls` check is only performed once, when the operator presses whatever button is mapped to `shoot`. Instead, the operator pressing/releasing their button should change an internal state of the Ball subsystem enabling it to shoot, and then the task here would implement control loop logic for checking if the shooter is ready and using the feeder to advance a ball when it is.

Reply:
Do you mean something like 
- operator presses button
- enables the task in the ball subsystem
- the task runs the shooter
- the task checks the RPM and then feeds the balls if the RPM is high enough

Reply:
Yea, basically.

Robot.java
```
void teleopSetup() {
  gamepad.bind(somebutton, ball::shoot);
  gamepad.bind(someOtherButton, ball::stopShooting);
}
```

Ball.java
```
interface Ball {
  void shoot();
  void cancelShooting();
}
```

BallImpl.java
```
class BallImpl implements Ball {
  private boolean shootEnabled;

  void shoot() {
    this.shootEnabled = true;
  }

  void cancelShooting() {
    this.shootEnabled = false;
  }

  void task() {
    if (this.shootEnabled) {
      // check shooter state, start it if required, feed it if ready
    }
  }
```

Reply:
When it comes to binding the controls I will look at it at a later date, I need to think about how I'm going to structure it.